### ISSUE TITLE
Why not middleware?

### ISSUE THREAD

**Issue by piotr-cz** [rt2zz/redux-persist-crosstab#19]
What it the benefit of implementing such functionality at the Redux-Persist level vs as Redux middleware (ie. like with [redux-state-sync](https://github.com/AOHUA/redux-state-sync))?

Only one I can think of is that as it's dependent on Redux-Persist, so all app instances are starting with latest state.

When similar functionality is implemented as middleware, only single actions are synced. Using this module whole state is being retrieved and hydrated.

**Comment by behnammodi** [comment]
Of course, `redux-persist-crosstab` is not a middleware only. `initStateWithPrevTab` and `withReduxStateSync` is Redux level method
